packages/wrangler/src/__tests__/publish.test.ts

```diff
@@ -1999,6 +2000,111 @@ addEventListener('fetch', event => {});`
       `);
     });
 
+    it("should batch assets in groups <100 mb", async () => {
+      // Let's have 20 files, from size 1 - 20 mb
+      const assets = Array.from({ length: 20 }, (_, index) => ({
+        filePath: `assets/file-${`${index}`.padStart(2, "0")}.txt`,
+        content: "X".repeat(1024 * 1024 * (index + 1)),
+      }));
+
+      const kvNamespace = {
+        title: "__test-name-workers_sites_assets",
+        id: "__test-name-workers_sites_assets-id",
+      };
+      writeWranglerToml({
+        main: "./index.js",
+        site: {
+          bucket: "assets",
+        },
+      });
+      writeWorkerSource();
+      writeAssets(assets);
+      mockUploadWorkerRequest();
+      mockSubDomainRequest();
+      mockListKVNamespacesRequest(kvNamespace);
+      mockKeyListRequest(kvNamespace.id, []);
+
+      // We expect this to be uploaded in 3 batches
+      // The first batch has 11 files (88mb)
+      // The next batch has 5 files (93mb)
+      // And the last one has 4 files (98mb)
+
+      // Since the bulk upload api endpoint stays the same
+      // We're going to have to clear the mock as soon as it's resolved
+      // And immediately add a mock for another one
+      // Welcome to a callback pyramid in 2022
```

Comment:
I think we could make this a bit less painful if we capture useful stuff in the handler and then check it after `runWrangler` has completed...


```
const requests = mockUploadAssetsToKVRequest(kvNamespace.id);
await runWrangler("publish);
expect(requests.uploads).toEqual(expectedAssets);
```

Either write out the `expectedAssets` matchers by hand, or generate them based on the `assets` array.

We would need to tweak the `mockUploadAssetsToKVRequest()` so that you don't need to specify `assets` as a parameter, and instead add them to a `requests` object that should be returned from the call.

Reply:
I've added a follow-up task for this as it'll require refactoring the other tests too: https://github.com/cloudflare/wrangler2/issues/1245